python-checks/src/test/resources/checks/hotspots/graphQLIntrospection.py

```diff
@@ -0,0 +1,269 @@
+        validation_rules=[OtherSensitiveRule, some_rule]
+    )
+
+    GraphQLView.as_view( 
+        name="introspection",
+        schema=schema,
+        graphiql=True,
+        validation_rules=[introspection_rule]
+    )
+
+
+def graphql_server_middleware_non_compliant(schema, some_middleware):
+    from graphql_server.flask import GraphQLView
+
+    GraphQLView.as_view(  # Noncompliant {{Disable introspection on this "GraphQL" server endpoint.}}
+#   ^^^^^^^^^^^^^^^^^^^
+        name="introspection",
+        schema=schema,
+        graphiql=True,
+    )
+
+    GraphQLView.as_view(  # Noncompliant 
+#   ^^^^^^^^^^^^^^^^^^^
+        name="introspection",
+        schema=schema,
+        graphiql=True,
+        middleware=[]
+    )
+
+    GraphQLView.as_view(  # Noncompliant
+#   ^^^^^^^^^^^^^^^^^^^
+        name="introspection",
+        schema=schema,
+        graphiql=True,
+        middleware=[some_middleware]
+    )
+
+
+def graphql_server_middleware_compliant(schema, introspection_middleware, get_introspection_middlewares):
+    from graphql_server.flask import GraphQLView
+
+    GraphQLView.as_view( 
+        name="introspection",
+        schema=schema,
+        graphiql=True,
+        middleware=[introspection_middleware],
+    )
+
+    GraphQLView.as_view(
+        name="introspection",
+        schema=schema,
+        graphiql=True,
+        middleware=get_introspection_middlewares(),
+    )
+
+
+def graphql_server_validation_rules_non_compliant(schema, some_rule):
+    from graphql_server.flask import GraphQLView
+    import graphene
+
+    GraphQLView.as_view(  # Noncompliant 
+#   ^^^^^^^^^^^^^^^^^^^
+        name="introspection",
+        schema=schema,
+        graphiql=True,
+        validation_rules=[]
+    )
+
+    GraphQLView.as_view(  # Noncompliant 
+#   ^^^^^^^^^^^^^^^^^^^
+        name="introspection",
+        schema=schema,
+        graphiql=True,
+        validation_rules=[some_rule]
+    )
+
+
+def graphql_server_validation_rules_compliant(schema, some_rule, introspection_rule):
+    from graphql_server.flask import GraphQLView
+    import graphene
+    from graphene.validation import DisableIntrospection
+    from graphene.validation import DisableIntrospection as SafeRule
+    from graphql.validation import NoSchemaIntrospectionCustomRule
+
+    GraphQLView.as_view(
+        name="introspection",
+        schema=schema,
+        graphiql=True,
+        validation_rules=[NoSchemaIntrospectionCustomRule, some_rule]
+    )
+
+    GraphQLView.as_view(
+        name="introspection",
+        schema=schema,
+        graphiql=True,
+        validation_rules=[DisableIntrospection, some_rule]
+    )
+
+    GraphQLView.as_view(
+        name="introspection",
+        schema=schema,
+        graphiql=True,
+        validation_rules=[SafeRule, some_rule]
+    )
+
+    GraphQLView.as_view( 
+        name="introspection",
+        schema=schema,
+        graphiql=True,
+        validation_rules=[introspection_rule]
+    )
```

Comment:
Maybe it would also be useful to have a test case where we have both of the keyword arguments in the `as_view()` function call, and combine cases of these arguments containing compliant and non_compliant classes and methods?